rest-api-spec/src/main/resources/rest-api-spec/test/termvectors/50_mix_typeless_typeful.yml

```diff
@@ -0,0 +1,45 @@
+---
+"Term vectors with typeless API on an index that has types":
+
+ - skip:
+      version: " - 6.99.99"
+      reason:  Typeless APIs were introduced in 7.0.0
+
+ - do:
+      indices.create: # not using include_type_name: false on purpose
+          index: index
+          body:
+              mappings:
+                  not_doc:
+                      properties:
+                          foo:
+                              type: "text"
+                              term_vector: "with_positions"
+
+ - do:
+      index:
+          index:  index
+          type:   not_doc
+          id:     1
+          body:   { foo: bar }
+
+ - do:
+      indices.refresh: {}
+
+ - do:
+      termvectors:
+          index: index
+          type:  _doc # todo: remove when termvectors support typeless API
+          id:    1
+
+ - is_true: found
+ - match: {_type: _doc}
+ - match: {term_vectors.foo.terms.bar.term_freq: 1}
+
+ - do:
+      termvectors:
+          index: index
+          type:  some_random_type
+          id:    1
+
+ - is_false: found 
```

Comment:
do we also need to check what happens when providing not_doc as a type?

Reply:
we have one already